### ISSUE TITLE
[Android] Name assert: Attempted to construct name literal on a different thread from the first initialized static name

### ISSUE THREAD

**Issue by moraaar** [o3de/o3de#10608]
**Describe the bug**
Launching a default level on Android the following error appears in the log:

````
2022-07-11 11:46:33.148 6521-6560/org.o3de.OpenXRTest I/System: Trace::Assert
     E:/Amazon/GitHub/o3de/Code/Framework/AzCore/AzCore/Name/Name.cpp(207): (517333306544) 'void AZ::Name::SetNameLiteral(AZStd::string_view)'
2022-07-11 11:46:33.148 6521-6560/org.o3de.OpenXRTest I/System: Attempted to construct name literal on a different thread from the first initialized static name, this is unsafe
````

After modifying the assert to print more information, it indicated it's the name **"-fixed name dictionary deferred head-"** what has issues with:

````
2022-07-11 11:46:33.148 6521-6560/org.o3de.OpenXRTest I/System: Trace::Assert
     E:/Amazon/GitHub/o3de/Code/Framework/AzCore/AzCore/Name/Name.cpp(207): (517333306544) 'void AZ::Name::SetNameLiteral(AZStd::string_view)'
2022-07-11 11:46:33.148 6521-6560/org.o3de.OpenXRTest I/System: Attempted to construct name literal '-fixed name dictionary deferred head-' on a different thread (517333306544) from the first initialized static name (532865967360), this is unsafe
````

**Steps to reproduce**
1. Run default level on Android

**Expected behavior**
No asserts happen when running a default level on Android

**Actual behavior**
The name **"-fixed name dictionary deferred head-"** causes an assert because it's being initialized from different threads.

**Found in Branch**
Development 5801ff7ad1beca136eb28a16ce90bd16694d2a80

**Desktop/Device (please complete the following information):**
 - Device: [Samsung] 
 - OS: [Android]
 - Version [12]

**Comment by moraaar** [comment]
With the latest code from development (cc6c096) now this causes an instant crash on Android. Increasing to blocker since android development is blocked by this.

The callstack shows it comes from this line:
````
const Name Handler = Name::FromStringLiteral("Handler", AZ::Interface<AZ::NameDictionary>::Get());
````

At the end it's crashing while calling `APKFileHandler& apkHandler = Get();` in `APKFileHandler::SetNumBytesToRead` while trying to read "/proc/self/status" in `bool performDebuggerDetection()`

This could be related to the recent changes to the Name system https://github.com/o3de/o3de/pull/10986 (Tagging @lumberyard-employee-dm)

![image (10)](https://user-images.githubusercontent.com/27999040/183432583-6339af0b-f72a-43db-afe3-5636ee52e02f.png)

**Comment by moraaar** [comment]
This issue still occurs with the latest (a8199a8f34231c9f23b22b077c14fe5553be3f28) that has the Interface fix: https://github.com/o3de/o3de/pull/11176

**Comment by lumberyard-employee-dm** [comment]
I think we can remove the NameLiteral initialized on different thread [assert](https://github.com/o3de/o3de/blob/development/Code/Framework/AzCore/AzCore/Name/Name.cpp#L231) because at static initialization time, there is only one thread.

The use case for it seems to be trying to assert if at runtime, after the Dictionary has been created, any additional NameLiterals are only created on the thread the first NameLiteral was initialized on for that specific module.
That API doesn't provide any meaningful functionality.

If anything it is the linking logic to the NameLiteral LinkedList that should be using atomics, to update the entries in Name::LinkStaticName and Name::UnlinkStaticName https://github.com/o3de/o3de/blob/development/Code/Framework/AzCore/AzCore/Name/Name.cpp#L262-L302
